brianlong/www.validators.app
# 186442305 save validator images

#### What's this PR do?
- save validator avatars as image files and display it instead of avatar uls

#### How should this be manually tested?
- install ImageMagick
- run bundle install
- run migrations
- run tests
- start sidekiq
- get avatar_urls by `rails r script/validators_get_info.rb` (cli) and `rails r script/validators_get_avatar_url.rb` (keybase)
- download files for existing urls by  `rails r script/update_validator_avatars.rb`
- check logs at log/update_avatar_file_service.log
- check if avatars are attached to validators with avatar_url ( i.e. validator.avatar.attached? )
- check if avatars displayed for validators on pages like validators index, details etc are grabbed from storage

2nd Scenario:
- Change avatar_url link for one of the validators to test new links detection
- Confirm that the job was created
- After the job is completed, confirm that the url was updated and file downloaded and attached

3rd scenario:
- Test files presence in API responses

#### What are the relevant tickets?
- [URL to the PivotalTracker ticket](https://www.pivotaltracker.com/story/show/186442305)

#### Task completed checklist:
- [] Is there appropriate test coverage?
- [] Did I take into consideration possible security vulnerabilities?
- [] Are the controllers secure?
- [] Was every important matter documented?

#### Things to watch out for
- [Y] Does this PR have any migrations?
- [Y] Does this PR add new dependencies?
